vhdx/src/vhdx.rs

```diff
@@ -0,0 +1,73 @@
+mod vhdx_header;
+mod vhdx_metadata;
+
+use crate::vhdx_header::{RegionTableEntry, VhdxHeader};
+
+use std::collections::btree_map::BTreeMap;
+use std::fmt::{self, Display};
+use std::fs::File;
+
+pub enum Error {
+    VhdxHeaderParsingFailed(vhdx_header::Error),
+    VhdxRegionEntryParsingFailed(vhdx_header::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+impl Display for Error {
+    #[remain::check]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::Error::*;
+
+        #[sorted]
+        match self {
+            VhdxHeaderParsingFailed(e) => write!(f, "Failed to parse VHDx Header {}", e),
+            VhdxRegionEntryParsingFailed(e) => write!(f, "Failed to parse VHDx Header {}", e),
+        }
+    }
+}
+
+const REGION_SIZE: u64 = 64 * 1024; // Each header size if 64KB
+const REGION1_START: u64 = 192 * 1024; // Region 1 start in Bytes
+const REGION2_START: u64 = 256 * 1024; // Region 2 start in Bytes
+const REGION1_ENTRY_START: u64 = REGION1_START + REGION_SIZE;
+const REGION2_ENTRY_START: u64 = REGION2_START + REGION_SIZE;
```

Comment:
This is wrong, right? The beginning of the region entries is only 16bytes after the beginning of each region.
What you wrote means that `REGION1_ENTRY_START` is the same as `REGION2_START`.